File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] nums, int k, int threshold) {
        int left = 0;
        int right = 0;
        int currentSum = 0;
        int currentAverage = 0;
        int result = 0;
        while(right < nums.length)
        {
            if(right - left + 1 < k)
            {
                currentSum += nums[right];
                right++;
                continue;
            }

            currentSum += nums[right];
            currentAverage = currentSum / k;
            if(currentAverage >= threshold) result++;
            currentSum -= nums[left];
            left++;
            right++;
        }
        return result;
    }
}

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.go
func numOfSubarrays(arr []int, k int, threshold int) int {
    left := 0
    count := 0
    currentTotal := 0

    for right := 0; right < len(arr); right++ {
        currentTotal = currentTotal + arr[right]

        if right - left + 1 > k {
            currentTotal = currentTotal - arr[left]
            left++
        }

        if right - left + 1 == k && currentTotal / k >= threshold {
            count++
        }

    }
    return count
}